pkg/security/probe/process_killer.go

```diff
@@ -257,7 +260,7 @@ func (p *ProcessKiller) updateKillQueueAlarmOnDisarm(disarmer *ruleDisarmer) {
 	// the correct value. Disable the alarm otherwise.
 	// NB: we should not kill anything, except if there is a race between a rule
 	//     disarmament and a kill queue alarm
-	p.KillQueuedPidsAndSetNextAlarm()
+	p.killQueuedPidsAndSetNextAlarm()
```

Comment:
Are you sure all callers of `updateKillQueueAlarmOnDisarm` are locking ruleDisarmersLock ? I'm not seeing that in the code

Reply:
Indeed none of the callers are locking `ruleDisarmersLock`, this was already the case before and is definitely an issue. I cannot simply take a lock `ruleDisarmersLock` because `disarmer.Lock` is already held here (and other code paths are using the `ruleDisarmersLock` -> `disarmer.Lock` order. Let me see how to fix that